commons/commons-util/src/main/kotlin/ebi/ac/uk/io/FileUtils.kt

```diff
@@ -123,14 +128,27 @@ object FileUtils {
 
     fun size(file: File): Long = Files.size(file.toPath())
 
-    fun md5(file: File): String = if (file.isFile) DigestUtils.md5Hex(file.readBytes()).toUpperCase() else ""
+    fun md5(file: File): String = if (file.isFile) calculateMd5(file).toUpperCase() else ""
 
     fun listFiles(file: File): List<File> =
         if (isDirectory(file)) Files.list(file.toPath()).map { it.toFile() }.toList() else emptyList()
 
     fun setFolderPermissions(path: Path, permissions: Set<PosixFilePermission>) {
         Files.setPosixFilePermissions(path, permissions)
     }
+
+    private fun calculateMd5(file: File): String = file.inputStream().use { inputStream ->
+        var read: Int
+        val buffer = ByteArray(BUFFER_SIZE)
+        val digest = MessageDigest.getInstance(MD5_ALGORITHM)
+
+        while (inputStream.read(buffer).also { read = it } > 0) {
+            digest.update(buffer, 0, read)
+        }
+
+        val md5sum = BigInteger(CHECKSUM_SIGNUM, digest.digest()).toString(HEXADECIMAL_BASE)
+        return String.format("%32s", md5sum).replace(" ", "0")
+    }
```

Comment:
```
  fun md5(file: File): String {
        val digest = MessageDigest.getInstance(MD5_ALGORITHM)
        file.inputStream().buffered(BUFFER_SIZE).iterator().forEach { digest.update(it) }
        return digest.digest().joinToString("") { "%02x".format(it) }
    }
```

Maybe?

Reply:
Works like a charm! I knew there was a better way to write this. Nice comment 😁👍

Reply:
Fixed